commit 93faba4505b2c02c4885d33f785db97efd6f9c1b
Author: nneul at neulinger.org <nneul@neulinger.org>
Date:   Fri Oct 6 12:08:58 2017 +0000

    Update log msg for Invalid Destination and Invalid Audience to show both the expected and received values.

diff --git a/auth_mellon_handler.c b/auth_mellon_handler.c
index afb3195..030abe7 100644
--- a/auth_mellon_handler.c
+++ b/auth_mellon_handler.c
@@ -1461,8 +1461,8 @@ static int am_validate_conditions(request_rec *r,
 
         if (ar->Audience == NULL || strcmp(ar->Audience, providerID)) {
             ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r,
-                          "Invalid Audience in Conditions. Should be: %s",
-                          providerID);
+                          "Invalid Audience in Conditions. Should be '%s', but was '%s'",
+                          providerID, ar->Audience ? ar->Audience : "");
             return HTTP_BAD_REQUEST;
         }
     }
@@ -1791,8 +1791,8 @@ static int am_handle_reply_common(request_rec *r, LassoLogin *login,
     if (response->parent.Destination) {
         if (strcmp(response->parent.Destination, url)) {
             ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r,
-                          "Invalid Destination on Response. Should be: %s",
-                          url);
+                          "Invalid Destination on Response. Should be '%s', but was '%s'",
+                          url, response->parent.Destination);
             lasso_login_destroy(login);
             return HTTP_BAD_REQUEST;
         }
